ksql-engine/src/main/java/io/confluent/ksql/util/KafkaTopicClientImpl.java

```diff
@@ -107,12 +123,69 @@ public void deleteTopics(List<String> topicsToDelete) {
       }
     }
     if (!failList.isEmpty()) {
-      throw new KsqlException("Failed to clean up topics: " + failList.stream().collect(Collectors
-                                                                                          .joining(",")));
+      throw new KsqlException("Failed to clean up topics: " + failList.stream()
+          .collect(Collectors.joining(",")));
+    }
+  }
+
+  @Override
+  public void deleteInternalTopics(final String applicationId) {
+    if (!isDeleteTopicEnabled) {
+      log.warn("Cannot delete topics since 'delete.topic.enable' is false. ");
+      return;
+    }
+    try {
+      Set<String> topicNames = listTopicNames();
+      List<String> internalTopics = new ArrayList<>();
+      for (String topicName: topicNames) {
+        if (isInternalTopic(topicName, applicationId)) {
+          internalTopics.add(topicName);
+        }
+      }
+      if (!internalTopics.isEmpty()) {
+        deleteTopics(internalTopics);
+      }
+    } catch (Exception e) {
+      log.error("Exception while trying to clean up internal topics for application id: {}.",
+                applicationId, e);
+    }
+  }
+
+  private void init() {
+    try {
+      DescribeClusterResult describeClusterResult = adminClient.describeCluster();
+      List<Node> nodes = new ArrayList<>(describeClusterResult.nodes().get());
+      if (!nodes.isEmpty()) {
+        ConfigResource resource = new ConfigResource(ConfigResource.Type.BROKER,
+                                                     String.valueOf(nodes.get(0).id()));
+        DescribeConfigsResult
+            describeConfigsResult = adminClient.describeConfigs(Collections.singleton(resource));
+        Map<ConfigResource, Config> config = describeConfigsResult.all().get();
+
+        this.isDeleteTopicEnabled = config.get(resource)
+            .entries()
+            .stream()
+            .anyMatch(configEntry -> configEntry.name().equalsIgnoreCase("delete.topic.enable"));
```

Comment:
Don't we need to check that the value is 'true' as well?

Reply:
Good point, I was indeed checking it in the previous version. Reverting back :)